### PR TITLE
DRY up labels in ui plugins

### PR DESCRIPTION
**Pull request by mscolnick**
There shouldn't be any visual changes. But some components are spaced a bit better (i.e. using gap instead of margins).

I used this snippet to update the `construct_element` cell in  `marimo tutorial ui` to compare:
```
def construct_element(value):
    if value == mo.ui.array:
        return mo.ui.array([mo.ui.text(), mo.ui.slider(1, 10), mo.ui.date()])
    elif value == mo.ui.batch:
        return mo.md(
            """
            - Name: {name}
            - Date: {date}
            """
        ).batch(name=mo.ui.text(label="something"), date=mo.ui.date(label="something"))
    elif value == mo.ui.button:
        return mo.ui.button(
            value=0, label="click me", on_click=lambda value: value + 1
        )
    elif value == mo.ui.checkbox:
        return mo.ui.checkbox(label="check me")
    elif value == mo.ui.date:
        return mo.ui.date(label="something")
    elif value == mo.ui.dictionary:
        return mo.ui.dictionary(
            {
                "slider": mo.ui.slider(1, 10, label="something"),
                "text": mo.ui.text("type something!"),
                "array": mo.ui.array(
                    [
                        mo.ui.button(value=0, on_click=lambda v: v + 1)
                        for _ in range(3)
                    ],
                    label="buttons",
                ),
            }
        )
    elif value == mo.ui.dropdown:
        return mo.ui.dropdown(["a", "b", "c"], label="something")
    elif value == mo.ui.file:
        return [mo.ui.file(kind="button", label="something"), mo.ui.file(kind="area", label="something")]
    elif value == mo.ui.form:
        return mo.ui.text_area(placeholder="...", label="something").form()
    elif value == mo.ui.multiselect:
        return mo.ui.multiselect(["a", "b", "c"], label="something")
    elif value == mo.ui.number:
        return mo.ui.number(start=1, stop=10, step=0.5, label="something")
    elif value == mo.ui.radio:
        return mo.ui.radio(["a", "b", "c"], value="a", label="something")
    elif value == mo.ui.slider:
        return mo.ui.slider(start=1, stop=10, step=0.5, label="something")
    elif value == mo.ui.switch:
        return mo.ui.switch(label="something")
    elif value == mo.ui.table:
        return mo.ui.table(
            data=[
                {"first_name": "Michael", "last_name": "Scott"},
                {"first_name": "Dwight", "last_name": "Schrute"},
            ],
            label="Employees",
        )
    elif value == mo.ui.text:
        return mo.ui.text(label="something")
    elif value == mo.ui.text_area:
        return mo.ui.text_area(label="something")
    return None
```

### CODE REVIEW

File: frontend/src/plugins/impl/SliderPlugin.tsx [hunk 1]
```diff
@@ -54,16 +51,9 @@ const SliderComponent = (props: SliderProps): JSX.Element => {
...
   const id = useId();
-  const labelElement =
-    props.label === null ? null : (
-      <Label htmlFor={id}>
-        <HtmlOutput html={props.label} inline={true} />
-      </Label>
-    );
 
   return (
-    <div className={clsx(labelStyles.labelContainer, "mb-2")}>
```
Comment (mscolnick) [inline frontend/src/plugins/impl/SliderPlugin.tsx:65]: this bottom margin was removed too. i think its a bit more composable without an enforced margin, but lmk if this breaks some UI

File: frontend/src/plugins/impl/SwitchPlugin.tsx [hunk 2]
```diff
@@ -30,29 +27,20 @@ const SwitchComponent = ({
...
-        labelStyles.labelContainer,
-        "align-middle",
-        "items-start",
-        "gap-x-2"
-      )}
+    <Labeled
+      label={data.label}
+      align="right"
+      id={id}
+      labelClassName="text-md ml-1"
```
Comment (akshayka) [inline frontend/src/plugins/impl/SwitchPlugin.tsx:36]: - let's update the font size to match the other labels, i think that just means removing "text-md"
- let's put the label on the left to match the other plugins?

(These issues may have already existed, but might as well fix them now)

File: frontend/src/plugins/impl/TextAreaPlugin.tsx [hunk 3]
```diff
@@ -40,21 +38,10 @@ interface TextAreaComponentProps extends Data {
...
-    props.label === null ? null : (
-      <Label className="text-md">
-        <HtmlOutput html={props.label} />
-      </Label>
-    );
-
   return (
-    <div className={labelStyles.labelContainerBlock}>
-      {labelElement}
+    <Labeled label={props.label} align="top" labelClassName="text-md">
```
Comment (akshayka) [inline frontend/src/plugins/impl/TextAreaPlugin.tsx:42]: Let's remove `text-md` (sorry, I know this issue already existed)

File: frontend/src/plugins/impl/SwitchPlugin.tsx (hunk 2 above)
Comment (mscolnick) [reply frontend/src/plugins/impl/SwitchPlugin.tsx:36]: i think "controls" (i.e. checkbox and switch) should be on the right side. but we can add an option to expose "align"

**Review by akshayka — APPROVED** [review]

**Pull request merged**